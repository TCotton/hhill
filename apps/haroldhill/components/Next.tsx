import React, { MouseEvent, useEffect, useState } from 'react'
import Link from 'next/link'
import { getApiRoot } from 'nextjs-url'
import fetchWithErrorHandle from '../helpers/fetchWithErrorHandle'

function useResults(articleId) {
  const [results, setResults] = useState(null)
  useEffect(() => {
    let ignore = false
    const apiRoot = getApiRoot().href
    fetchWithErrorHandle(
      `${apiRoot}/navigationArticles?articleId=` + articleId + '&direction=next'
    ).then((nextArticle) => {
      if (nextArticle.message === 'ok' && !ignore)
        setResults(nextArticle?.result?.fullSlug)
    })
    return () => {
      ignore = true
    }
  }, [articleId])
  return results
}

const Previous = (props) => {
  const { articleId } = props
  const next = useResults(articleId)
  interface IMyLinkRefProps {
    onClick?: MouseEvent<HTMLAnchorElement>
    href?: string
  }

  type Ref = HTMLAnchorElement

  const NextLink = React.forwardRef<Ref, IMyLinkRefProps>(
    ({ onClick, href }, ref) => {
      return (
        <p>
          <a href={href} ref={ref} className={`govuk-forward-link`}>
            {/* eslint-disable-next-line react/no-unescaped-entities */}
            Next
          </a>
        </p>
      )
    }
  )
  NextLink.displayName = 'NextLink'

  return (
    <>
      {next && (
        <Link href={next} passHref legacyBehavior>
          <NextLink />
        </Link>
      )}
    </>
  )
}

export default React.memo(Previous)
