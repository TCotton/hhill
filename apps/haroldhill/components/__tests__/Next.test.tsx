import Next from '../Next'
import { render } from '@testing-library/react'
describe('Next', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<Next />)
    expect(baseElement).toBeTruthy()
  })
})
