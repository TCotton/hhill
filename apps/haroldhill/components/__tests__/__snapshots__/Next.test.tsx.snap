// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Next should match snapshot 1`] = `
<body>
  <div />
</body>
`;
